src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -736,13 +735,24 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     }
 
     val callActorName = s"CallActor-${workflow.id}-${callKey.tag}"
-    val callActorProps = callKey match {
-      case backendCallKey: BackendCallKey => CallActor.props(backendCallKey, locallyQualifiedInputs, backend, workflow)
-      case finalCallKey: FinalCallKey => CallActor.props(finalCallKey)
+    val futureCallActorProps = callKey match {
+      case backendCallKey: BackendCallKey =>
+        Future.successful(CallActor.props(backendCallKey, locallyQualifiedInputs, backend, workflow))
+      case finalCallKey: FinalCallKey =>
+        WorkflowMetadataBuilder.workflowMetadata(workflow.id, backend) map {
```

Comment:
Yes, yes it is. But I couldn't figure out an easier/simpler way to have a common interface for FinalCallActors that accepted different objects depending on the various types of FinalCallActor.